phoenix-core/src/main/java/org/apache/phoenix/monitoring/GlobalMetricRegistriesAdapter.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.hadoop.metrics2.lib.Interns;
+import org.apache.hadoop.metrics2.lib.MutableHistogram;
+import org.apache.hadoop.metrics2.source.JvmMetrics;
+
+/**
+ * Contents mostly copied from GlobalMetricRegistriesAdapter class from hbase-hadoop2-compat
+ * The adapter attaches HBase's MetricRegistry to Hadoop's DefaultMetricsSystem
+ * Doesn't handle dynamic attach/detach of registries
+ */
+public class GlobalMetricRegistriesAdapter {
+
+    private static final Log LOG = LogFactory.getLog(GlobalMetricRegistriesAdapter.class);
+    private static GlobalMetricRegistriesAdapter INSTANCE;
+
+    private GlobalMetricRegistriesAdapter() {
+        DefaultMetricsSystem.initialize("Phoenix");
+        JvmMetrics.initSingleton("Phoenix", "");
+    }
+
+    public static GlobalMetricRegistriesAdapter getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new GlobalMetricRegistriesAdapter();
+        }
+        return INSTANCE;
+    }
+
+    public void registerMetricRegistry(MetricRegistry registry) {
+        if (registry == null) {
+            LOG.warn("Registry cannot be registered with Hadoop Metrics 2 since it is null.");
+            return;
+        }
+
+        HBaseMetrics2HadoopMetricsAdapter adapter = new HBaseMetrics2HadoopMetricsAdapter(registry);
+        registerToDefaultMetricsSystem(registry, adapter);
+    }
+
+    private void registerToDefaultMetricsSystem(MetricRegistry registry, HBaseMetrics2HadoopMetricsAdapter adapter) {
+        MetricRegistryInfo info = registry.getMetricRegistryInfo();
+        LOG.info("Registering " + info.getMetricsJmxContext() + " " + info.getMetricsDescription() + " into DefaultMetricsSystem");
+        DefaultMetricsSystem.instance().register(info.getMetricsJmxContext(), info.getMetricsDescription(), adapter);
+    }
+
+    /**
+     * Class to convert HBase Metric Objects to Hadoop Metrics2 Metric Objects
+     */
+    private static class HBaseMetrics2HadoopMetricsAdapter implements MetricsSource {
+        private static final Log LOG = LogFactory.getLog(HBaseMetrics2HadoopMetricsAdapter.class);
+        private final MetricRegistry registry;
+        private HBaseMetrics2HadoopMetricsAdapter(MetricRegistry registry) {
+            this.registry = registry;
+        }
+
+        public void snapshotAllMetrics(MetricRegistry metricRegistry, MetricsCollector collector) {
+            MetricRegistryInfo info = metricRegistry.getMetricRegistryInfo();
+            MetricsRecordBuilder builder = collector.addRecord(Interns.info(info.getMetricsName(), info.getMetricsDescription()));
+            builder.setContext(info.getMetricsContext());
+            this.snapshotAllMetrics(metricRegistry, builder);
+        }
+
+        public void snapshotAllMetrics(MetricRegistry metricRegistry, MetricsRecordBuilder builder) {
+            Map<String, Metric> metrics = metricRegistry.getMetrics();
+            Iterator i$ = metrics.entrySet().iterator();
```

Comment:
why do you need this?  i$

Reply:
Yes, ideally should not have used it here. I copied the code from a class file. Will update it.